statsmodels/base/model.py

```diff
@@ -510,6 +510,149 @@ def hess(params, *args):
+        if hasattr(res_constr.model, 'scale'):
+            # GLM problem, see #2399
+            res.model.scale = res._results.scale = res_constr.model.scale
+
+        if hasattr(res_constr, 'mle_retvals'):
+            res.mle_retvals = res_constr.mle_retvals
+            # not available for not scipy optimization, e.g. glm irls
+#             res.mle_retvals['fcall'] = res_constr.mle_retvals.get('fcall', np.nan)
+#             res.mle_retvals['iterations'] = res_constr.mle_retvals.get(
+#                                                             'iterations', np.nan)
+#             res.mle_retvals['converged'] = res_constr.mle_retvals['converged']
+            # overwrite all settings
+            #res.mle_retvals.update(res_constr.mle_retvals)  #not yet
+            res.mle_settings.update(res_constr.mle_settings)
+
+        res._results.params = params_full
+        if not hasattr(res._results, 'normalized_cov_param') or res._results.normalized_cov_param is None:
+            res._results.normalized_cov_params = np.zeros((k_params, k_params))
+        else:
+            res._results.normalized_cov_params[...] = 0
+
+        # TODO: sandwiches, add cov_params_default ?
+        # fanxy indexing requires integer attay
+        keep_index = np.array(keep_index)
+        res._results.normalized_cov_params[keep_index[:, None], keep_index] = res_constr.normalized_cov_params
+        k_constr = res_constr.df_resid - res._results.df_resid
+        if hasattr(res_constr, 'cov_params_default'):
+            res._results.cov_params_default = np.zeros((k_params, k_params))
+            res._results.cov_params_default[keep_index[:, None], keep_index] = res_constr.cov_params_default
+        if hasattr(res_constr, 'cov_type'):
+            res._results.cov_type = res_constr.cov_type
+            res._results.cov_kwds = res_constr.cov_kwds
+
+        res._results.keep_index = keep_index
+        res._results.df_resid = res_constr.df_resid
+        res._results.df_model = res_constr.df_model
+
+        res._results.k_constr = k_constr
+        res._results.results_constrained = res_constr
+
+        # special temporary workaround for RLM
+        # need to be able to override robust covariances
+        if hasattr(res.model, 'M'):
+            del res._results._cache['resid']
+            del res._results._cache['fittedvalues']
+            del res._results._cache['sresid']
+            cov = res._results._cache['bcov_scaled']
+            # inplace adjustment
+            cov[...] = 0
+            cov[keep_index[:, None], keep_index] = res_constr.bcov_scaled
+            res._results.cov_params_default = cov
+
+        return res
+
+    def _fit_collinear(self, atol=1e-14, rtol=1e-13, **kwds):
+
+        # ------ copied from PR #2380 remove when merged
+        x = self.exog
+        tol = atol + rtol * x.var(0)
+        r = np.linalg.qr(x, mode='r')
+        mask = np.abs(r.diagonal()) < np.sqrt(tol)
+        #idx_collinear = np.where(mask)[0]
```

Comment:
pls delete commented-out

Reply:
should be used in future, or recreated in fit_zeros